aws/asg-ebs/user-data.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+echo ECS_CLUSTER=${ecs_cluster_name} >> /etc/ecs/ecs.config
+
+# Install the NFS client utilities
+echo "user_data.sh: Installing nfs-utils"
+yum install -y nfs-utils
+
+# Mount the EFS file system
+echo "user_data.sh: Mounting EFS file system at ${mount_point}"
+mkdir -p ${mount_point}
+echo "${efs_dns_name}:/ ${mount_point} nfs4 nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=600,retrans=2,auto 0 0" >> /etc/fstab
+mount -a
+
+
+# Install the AWS CLI (needed to enable attaching the EBS volume)
+echo "user_data.sh: Installing the AWS CLI"
+yum install -y unzip
+curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+unzip -q awscliv2.zip
+./aws/install
+
+# Configure the AWS CLI
+echo "user_data.sh: Configuring the AWS CLI"
+mkdir ~/.aws
+echo "[default]"               > ~/.aws/config
+echo "region = ${aws_region}" >> ~/.aws/config
+
+echo "[default]"                                  > ~/.aws/credentials
+echo "aws_access_key_id = ${aws_access_key}"     >> ~/.aws/credentials
+echo "aws_secret_access_key = ${aws_secret_key}" >> ~/.aws/credentials
+
+
+# Get my EC2 instance ID
+# From https://stackoverflow.com/questions/625644/how-to-get-the-instance-id-from-within-an-ec2-instance
+# INSTANCE_ID=`curl http://169.254.169.254/latest/meta-data/instance-id`
```

Comment:
Do you want this other method for getting the instance ID here for later reference?

Reply:
No, I don't need it and removed it.